packages/components/src/components/teaser/teaser.stories.ts

```diff
@@ -0,0 +1,352 @@
+      axis: {
+        x: { type: 'attribute', name: 'inset' },
+        y: { type: 'attribute', name: 'orientation' }
+      },
+      args,
+      constants: [
+        {
+          type: 'attribute',
+          name: 'variant',
+          value: 'primary-100'
+        }
+      ]
+    });
+  }
+};
+
+/**
+ * Teaser with different media and content distribution values.
+ */
+
+export const DistributionRatio = {
+  parameters: { controls: { exclude: ['--distribution-media', '--distribution-content'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: {
+          type: 'template',
+          name: 'media and content distribution',
+          values: [
+            {
+              title: 'media: 66%, content: 33%',
+              value: '<div style="--distribution-media: 66%; --distribution-content: 33%;">%TEMPLATE%</div>'
+            },
+            {
+              title: 'media: 200px, content: 450px',
+              value: '<div style="--distribution-media: 200px; --distribution-content: 450px;">%TEMPLATE%</div>'
+            }
+          ]
+        }
+      },
+      args,
+      constants: [
+        {
+          type: 'template',
+          name: 'style',
+          value: '<div style="margin-bottom: 50px">%TEMPLATE%</div>'
+        },
+        {
+          type: 'attribute',
+          name: 'orientation',
+          value: 'horizontal'
+        },
+        {
+          type: 'attribute',
+          name: 'variant',
+          value: 'primary-100'
+        }
+      ]
+    });
+  }
+};
+
+/**
+ * Responsive teaser in different Breakpoints.
+ */
+
+export const Breakpoint = {
+  parameters: { controls: { exclude: 'orientation' } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: {
+          type: 'template',
+          name: 'style',
+          values: [
+            {
+              value: '<div style="width: 300px">%TEMPLATE%</div>',
+              title: 'grid sm'
+            },
+            {
+              value: '<div style="width: 500px">%TEMPLATE%</div>',
+              title: 'grid md'
+            }
+          ]
+        }
+      },
+      args,
+      constants: [
+        {
+          type: 'template',
+          name: 'style',
+          value: '<div style="margin-bottom: 40px">%TEMPLATE%</div>'
+        },
+        {
+          type: 'attribute',
+          name: 'orientation',
+          value: 'responsive'
+        },
+        {
+          type: 'attribute',
+          name: 'variant',
+          value: 'primary-100'
+        }
+      ]
+    });
+  }
+};
+
+/**
+ * Different headline sizes. It is also possible to use sd-link inside the <h> tag in the headline slot.
+ */
+
+export const Headline = {
+  parameters: { controls: { exclude: 'headline' } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: {
+          type: 'slot',
+          name: 'headline',
+          values: [
+            {
```

Comment:
Please define the more programmatic, that something like 
```js
['h1', 'h2', 'h3'].map((tag) => { return {value: `<${tag} slot="headline">${tag}</${tag}`, title: tag}})
```
or similar (untested)